xarray/tests/test_variable.py

```diff
@@ -112,6 +112,37 @@ def test_getitem_with_mask(self):
         assert_identical(v._getitem_with_mask([0, -1, 1], fill_value=-99),
                          self.cls(['x'], [0, -99, 1]))
 
+    def test_getitem_with_mask_fancy(self):
+        def one_test(key):
+            # key should be a dict
+            v = self.cls(['x', 'y', 'z'], np.random.randn(4, 5, 6))
+            actual = v._getitem_with_mask(key)
```

Comment:
This will need a `.compute()` after it. Currently the tests uses dask fail, because dask does not support assignment.